clients/clearly-defined/src/main/kotlin/Coordinates.kt

```diff
@@ -62,10 +62,12 @@ data class Coordinates(
         provider = Provider.fromString(parts[1]),
         namespace = parts[2].takeUnless { it == "-" },
         name = parts[3],
-        revision = parts.getOrNull(4)
+        revision = parts[4].takeUnless { it.isEmpty() }
     )
 
-    override fun toString() = listOfNotNull(type, provider, namespace ?: "-", name, revision).joinToString("/")
+    val strings: List<String> = listOf(type.toString(), provider.toString(), namespace ?: "-", name, revision.orEmpty())
```

Comment:
Would `components` be a more speaking name maybe?

Reply:
"components" reminds a bit too much of the `ComponentType` class, and I'd like to emphasize that these are the string representations of all (also the nullable) properties. I had `entries` before, but that did also not reflect that bit good enough for my taste. So `strings` pragmatically was the best idea that I had.